src/NHibernate/Engine/ActionQueue.cs

```diff
@@ -169,6 +187,10 @@ public void ExecuteInserts()
 		/// </summary>
 		public void ExecuteActions()
 		{
+			if (session.Factory.Settings.IsQueryCacheEnabled)
```

Comment:
1) `ExecuteInserts` (a few lines above) should call this as well.
2) It seems that it shall be called after execute

Reply:
Also `Execute(action)` can be called explicitly.

Reply:
Bummer. I had made a mental note of it, but forgot.